### PR TITLE
✨ Allow local audio files in background-audio

### PR DESCRIPTION
**Pull request by apatronl**
Removes https check to allow for local usage of background-audio property.

### CODE REVIEW

File: extensions/amp-story/1.0/audio.js [hunk 1]
```diff
@@ -33,10 +33,7 @@ export function upgradeBackgroundAudio(element, loop = true) {
     return null;
   }
   const audioEl = element.ownerDocument.createElement('audio');
-  const audioSrc = Services.urlForDoc(element).assertHttpsUrl(
-    element.getAttribute('background-audio'),
-    element
-  );
+  const audioSrc = element.getAttribute('background-audio');
```
Comment (processprocess) [inline extensions/amp-story/1.0/audio.js:36]: Hi @apatronl! Thanks for the contribution :) 

We do this check to be sure the audio file is being served securely through https. 
It looks like this commit would remove the check.

Local audio files can be tested while developing. The `assertHttpsUrl` provides an exception for localhost:
https://github.com/ampproject/amphtml/blob/main/src/service/url-impl.js#L120-L133
 
Please let me know if you have any other questions about the check and why it's necessary :)

**Comment by apatronl** [comment]
Thanks for your reply, @processprocess!

This change would be helpful to be able to use local files for previewing a story on mobile (i.e. create/edit + preview story), without having to use a local host. Local images and videos aren't blocked at runtime despite following similar HTTPS rules, it's just audio files that are blocked. This change also won't affect behavior on valid AMP pages.

Let me know what you think!

**Comment by gmajoulet** [comment]
I just chatted with Alan who's the components TL and he shared with me that this check is to make sure there's no mixed content security warning when served from the AMP Cache (always HTTPS).
If we remove this check then it becomes possible for publishers to have something that works locally but doesn't work on other platforms.

**Comment by choumx** [comment]
> this check is to make sure there's no mixed content security warning when served from the AMP Cache (always HTTPS).

Doesn't the mixed content warning also apply to images/videos (which doesn't have this behavior today)?

Alternatively: can we keep the user JS error as a developer warning for this, but modify the runtime exception so that audio isn't broken?

**Comment by gmajoulet** [comment]
> Doesn't the mixed content warning also apply to images/videos (which doesn't have this behavior today)?

Images always get rewritten to a HTTPS AMP Cache URL so they don't have this potential mixed content issue. Both `amp-video` and `amp-audio` have it though and only accept HTTPS url (with the exception of `localhost` for dev purposes)

> Alternatively: can we keep the user JS error as a developer warning for this, but modify the runtime exception so that audio isn't broken?
It'd only fix it for your origin testing, but be broken on any platform using the AMP Cache. Keeping errors consistent on origin and cache sounds like a good approach to me.
I didn't come up with this rule and I'm not the one enforcing it tho, maybe you can find someone from the Bento team to weigh in (cc @alanorozco).

**Comment by apatronl** [comment]
> Both `amp-video` and `amp-audio` have it though and only accept HTTPS url (with the exception of `localhost` for dev purposes)

Based on our current usage of `amp-video`, it doesn't look like it only accepts HTTPS/`localhost` urls. We've had no problem using local video files for story preview, though now I'm not sure whether this is a bug or a feature?

**Comment by gmajoulet** [comment]
`amp-video` uses the same helper that our background-audio you're trying to remove, cf [[1]](https://github.com/ampproject/amphtml/blob/ff26a3c602bea8fb673e2a038fd7cc3b66e4e087/extensions/amp-video/0.1/amp-video.js#L573) and [[2]](https://github.com/ampproject/amphtml/blob/ff26a3c602bea8fb673e2a038fd7cc3b66e4e087/extensions/amp-video/0.1/amp-video.js#L584).

The exact definition of what you can or cannot do is:
```
must start with "https://" or "//" or be relative and served from either https or from localhost.
```

But they all use the same helper, so what you can or cannot do is supposed to be consistent across background-audio, amp-audio, and amp-video.

**Comment by apatronl** [comment]
Thanks for the pointers, Gabriel!

I think maybe we've found a separate bug for `amp-video` when embedded in `amp-story-page`. When I test locally, I'm able to see locally stored videos in a story without starting a `localhost`; I do see the error message:
```
must start with "https://" or "//" or be relative and served from either https or from localhost.
```
But the `video` element is still added to the dom even with the https assertion failure.

E.g. Given the following code:

```
<!doctype html><html ⚡><head><meta charset="utf-8"><link href="TODO.html" rel="canonical"><meta content="width=device-width,minimum-scale=1" name="viewport"><style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript><style amp-custom>:root {--story-width-px:100 * var(--story-page-vw);--story-height-px:177.77777777777777 * var(--story-page-vw)}body {font-family:Arial;}amp-story-grid-layer * {position:absolute;}.safe_to_aspect {height:calc(1.1851852 * 100%);top:calc(-0.0925926 * 100%);}amp-story-grid-layer {overflow:visible;}</style><script async src="https://cdn.ampproject.org/v0.js"></script><script async custom-element="amp-story" src="https://cdn.ampproject.org/v0/amp-story-1.0.js"></script><title></title></head><body><amp-story poster-portrait-src="./TODO.jpeg" publisher-logo-src="TODO" publisher="TODO" standalone title>

<amp-story-page auto-advance-after="7s" id="page2">
  <amp-story-grid-layer aspect-ratio="2:3" template="vertical">
    <div class="safe_to_aspect">
      <amp-video id="1234" layout="fill" src="./snappingdemo.mov" autoplay="" poster="/TODO.jpg"></amp-video>

    </div>
  </amp-story-grid-layer>
</amp-story-page>
</amp-story></body></html>
```

You can see in the screenshot attached that the video is added even with the "invalid" source, and it is correctly displayed in my story.

amp-video

If the source is relative, maybe we should also allow file URLs in addition to HTTPS/localhost.

**Pull request closed without merging**